### PR TITLE
USHIFT-1801: change build_rpms.sh to create rpm with podman in CI

### PR DESCRIPTION
**Pull request by eslutsky**
this change will  use openshift's  go container toolchain in CI  ,make the resulting binary/rpm FIPs compliant and consistent with ART .

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -293,15 +293,14 @@ commit: image-build-configure image-build-commit
...
-	RPM_BUILDER_IMAGE_TAG="rhel-8-release-golang-1.19-openshift-4.13"; \
+	RPM_BUILDER_IMAGE_TAG="rhel-8-release-golang-1.20-openshift-4.14"; \
 	podman build \
 		--volume /etc/pki/entitlement/:/etc/pki/entitlement \
 		--build-arg TAG=$$RPM_BUILDER_IMAGE_TAG \
 		--tag microshift-builder:$$RPM_BUILDER_IMAGE_TAG - < ./packaging/images/Containerfile.rpm-builder ; \
 	podman run \
 		--rm -ti \
-		--volume $$(pwd):/opt/microshift \
-		--volume $$(go env GOCACHE):/go/.cache \
```
Comment (pmtk) [inline Makefile:304]: Have you checked how removing cache mount influences rebuild times?

Comment (eslutsky) [reply Makefile:304]: actually I've received `permission error` with that cache, did it work for you ?

File: test/bin/build_rpms.sh [hunk 2]
```diff
@@ -14,7 +14,7 @@ rm -rf _output/rpmbuild*
 
 BUILD_CMDS=( \
     # Normal build of current branch from source
-    'make rpm' \
+    'make rpm-podman' \
```
Comment (pmtk) [inline test/bin/build_rpms.sh:17]: This add new requirement to test harness. I think we should mention it in the docs

File: Makefile (hunk 1 above)
Comment (eslutsky) [reply Makefile:304]: looks like does improves the runtime in couple of minutes , i'll try adding `:z`  for selinux relabeling , like i did with the first volume.

File: test/bin/build_rpms.sh (hunk 2 above)
Comment (eslutsky) [reply test/bin/build_rpms.sh:17]: yes actually its requires:
1. pull-secret for the registry.ci.openshift.org 
2.  build has to run from a RHEL  RHSM registered machine

Comment (pmtk) [reply test/bin/build_rpms.sh:17]: FYI you could also copy entitlement from RHEL to Fedora and it should work. But for docs, let's keep it simple

File: Makefile [hunk 3]
```diff
@@ -57,6 +57,9 @@ GO_PACKAGES=$(go list ./cmd/... ./pkg/...)
 # Build to a place we can ignore
 GO_BUILD_BINDIR :=$(OUTPUT_DIR)/bin
 
+# Go Cache
```
Comment (pmtk) [inline Makefile:60]: Nit: I'm not sure we need this comment ;)

File: test/bin/build_rpms.sh (hunk 2 above)
Comment (ggiguash) [reply test/bin/build_rpms.sh:17]: Can/should we make this optional, only if registry.ci.openshift.org pull secret is present?

Comment (eslutsky) [reply test/bin/build_rpms.sh]: @ggiguash , i can do something different:
if the command `make rpm-podman` then fallback to `make rpm`

Comment (eslutsky) [reply test/bin/build_rpms.sh]: @pmtk  , now this change affect CI  only its not really affects the dev-env requirement. do we still need to change the Docs?

Comment (pmtk) [reply test/bin/build_rpms.sh]: Nah, I think this is fine

File: test/bin/build_rpms.sh [hunk 4]
```diff
@@ -12,9 +12,15 @@ source "${SCRIPTDIR}/common.sh"
 cd "${ROOTDIR}"
 rm -rf _output/rpmbuild*
 
-BUILD_CMDS=( \
-    # Normal build of current branch from source
-    'make rpm' \
+# Normal build of current branch from source
+BUILD_CMDS=('make rpm')
+
+#in CI build current branch from source using podman
```
Comment (ggiguash) [inline test/bin/build_rpms.sh:18]:
```suggestion
# In CI, build the current branch from source with the build tools used by OCP.
# This is necessary for enabling FIPS support.
```

**Pull request merged**